drivers/platform/x86/thinkpad-wmi.c

```diff
@@ -412,10 +444,18 @@ static int thinkpad_wmi_password_settings(struct thinkpad_wmi_pcfg *pcfg)
 	obj = output.pointer;
 	if (!obj || obj->type != ACPI_TYPE_BUFFER || !obj->buffer.pointer)
 		return -EIO;
+			
 	if (obj->buffer.length != sizeof(*pcfg)) {
-		pr_warn("Unknown pcfg buffer length %d\n", obj->buffer.length);
-		kfree(obj);
-		return -EIO;
+	   if (obj->buffer.length >sizeof(*pcfg)){
```

Comment:
can you add a few comments explaining why the size can be different ?

The c-style looks off, see https://www.kernel.org/doc/html/v4.10/process/coding-style.html